airflow/utils/dag_processing.py

```diff
@@ -308,6 +369,249 @@ def file_path(self):
+        # Map from file path to the number of runs
+        self._run_count = defaultdict(int)
+        # Pids of DAG parse
+        self._all_pids = []
+        # Pipe for communicating signals
+        self._parent_signal_conn, self._child_signal_conn = multiprocessing.Pipe()
+        # Pipe for communicating DagParsingStat
+        self._stat_queue = multiprocessing.Queue()
+        self._result_queue = multiprocessing.Queue()
+        self._process = None
+        self._done = False
+        # Initialized as true so we do not deactivate w/o any actual DAG parsing.
+        self._all_files_processed = True
+        self._result_count = 0
+
+    def start(self):
+        """
+        Launch DagFileProcessorManager processor and start DAG parsing loop in manager.
+        """
+        self._process = self._launch_process(self._dag_directory,
+                                             self._file_paths,
+                                             self._max_runs,
+                                             self._processor_factory,
+                                             self._child_signal_conn,
+                                             self._stat_queue,
+                                             self._result_queue,
+                                             self._async_mode)
+        self.log.info("Launched DagFileProcessorManager with pid: {}"
+                      .format(self._process.pid))
+
+    def heartbeat(self):
+        """
+        Should only be used when launched DAG file processor manager in sync mode.
+        Send agent heartbeat signal to the manager.
+        """
+        self._parent_signal_conn.send(DagParsingSignal.AGENT_HEARTBEAT)
+
+    def wait_until_finished(self):
+        """
+        Should only be used when launched DAG file processor manager in sync mode.
+        Wait for done signal from the manager.
+        """
+        while True:
+            if self._parent_signal_conn.poll() \
+                    and self._parent_signal_conn.recv() == DagParsingSignal.MANAGER_DONE:
+                break
+            time.sleep(0.1)
+
+    @staticmethod
+    def _launch_process(dag_directory,
+                        file_paths,
+                        max_runs,
+                        processor_factory,
+                        signal_conn,
+                        _stat_queue,
+                        result_queue,
+                        async_mode):
+        def helper():
+            # Reload configurations and settings to avoid collision with parent process.
+            # Because this process may need custom configurations that cannot be shared,
+            # e.g. RotatingFileHandler. And it can cause connection corruption if we
+            # do not recreate the SQLA connection pool.
+            os.environ['CONFIG_PROCESSOR_MANAGER_LOGGER'] = 'True'
+            reload(airflow.config_templates.airflow_local_settings)
+            reload(airflow.settings)
```

Comment:
This reload is done soley to reconfigure the logger right? Wouldn't it be more immediate and less "hacky" (using reload is a bit of a kludge) to reconfigure the logger instead?

Reply:
Or even better: could we use a different logger name and have it already configured with normal logging, but just only used by this class/logger?

Reply:
W/o reloading `airflow_local_settings`, the custom logic [here](https://github.com/apache/incubator-airflow/blob/c45cc19d87faa49cba1bbb4755940d3113adc522/airflow/config_templates/airflow_local_settings.py#L199-L203) would not be evaluated and thus we'll end up reconfigure with the same logging config again...

Also as mentioned in the comment, right now the major purposes of reloading `settings` is to reload logger and also the connection pool.

Tho I do agree reload is a bit of a kludge--I would avoid it if I had other solutions :(

Reply:
Can you do something like this?

```
logging.config.dictConfig({
  # New handler,
  disable_existing_loggers=False
}
```

Reply:
Nice trick here 👍 But moving the def of handler and logger here would prevent people from using there custom handler and logger for DAG parsing right?(e.g. logger with both `processor_manager` and `console` handler) Also we've recently encountered a rare case that same logger on multiple process would create lock issue if we do not reload and thus it feel safer to just reload all loggers.